arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/bookmark/BookmarkAdapter.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.esri.arcgisruntime.toolkit.bookmark
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.esri.arcgisruntime.mapping.Bookmark
+import com.esri.arcgisruntime.toolkit.R
+import kotlinx.android.synthetic.main.item_bookmark_row.view.*
+
+class BookmarkAdapter(
+    private val onItemClickListener: OnItemClickListener<Bookmark>,
+    diffCallback: DiffUtil.ItemCallback<Bookmark> = DiffCallback(),
+    private val itemLayoutRes: Int? = null
+) : ListAdapter<Bookmark, ViewHolder>(diffCallback) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        return ViewHolder(
+            inflater.inflate(
+                itemLayoutRes ?: R.layout.item_bookmark_row,
+                parent,
+                false
+            )
+        )
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) =
+        holder.bind(getItem(position), onItemClickListener)
+
+    class DiffCallback : DiffUtil.ItemCallback<Bookmark>() {
+
+        override fun areItemsTheSame(oldItem: Bookmark, newItem: Bookmark): Boolean {
+            return oldItem == newItem
+        }
+
+        override fun areContentsTheSame(oldItem: Bookmark, newItem: Bookmark): Boolean {
+            return oldItem.name == newItem.name && oldItem.viewpoint.toJson() == newItem.viewpoint.toJson()
+        }
+    }
+
+    interface OnItemClickListener<T> {
+        fun onItemClick(item: T)
+    }
+}
+
+class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+
+    fun bind(
+        bookmark: Bookmark,
+        onItemClickListener: BookmarkAdapter.OnItemClickListener<Bookmark>
+    ) {
+        itemView.txtBookmarkName.text = bookmark.name
```

Comment:
This only works if the layout used contains the `TextView` `txtBookmarkName`. You should be binding the `bookmark.name` property to the `TextView`

Reply:
if I bind the `bookmark.name` to the `TextView` it is less cleaner than this. I have made the change so you can have a look at how I am doing it. 
If the User want to override the layout then they can name the `TextView` in the layout as `txtBookmarkName` to make use of this binding, right ?

Reply:
But this isn't databinding in the same sense. The Adapter should allow the user to bind whatever properties from the Bookmark they wish to bind to their view. It just so happens that we're only binding the name here. By using a pure Data bound layout, we're showing the user how they can also do so.